File: bench/scenarios/polyglot-monorepo/packages/shared/src/normalize.ts
import type { OrderInput } from "../../api/src/routes/orders.ts";
import type { NormalizedOrder } from "./billing.ts";

export function normalizeOrderPayload(input: OrderInput): NormalizedOrder {
  return {
    orderId: input.orderId.trim(),
    customerId: input.customerId.trim(),
    amountCents: Math.max(0, input.amountCents),
  };
}

File: bench/scenarios/polyglot-monorepo/python/services/normalize.py
def normalize_order_payload_bridge(raw_payload: dict) -> dict:
    return {
        "order_id": str(raw_payload.get("order_id", "")).strip(),
        "customer_id": str(raw_payload.get("customer_id", "")).strip(),
        "amount_cents": int(raw_payload.get("amount_cents", 0)),
    }
